evalml/pipelines/component_graph.py

```diff
@@ -228,16 +229,16 @@ def _compute_features(self, component_list, X, y=None, fit=False):
                     elif isinstance(parent_x, ww.DataColumn):
                         parent_x = pd.Series(_convert_woodwork_types_wrapper(parent_x.to_series()), name=parent_input)
                     x_inputs.append(parent_x)
-            input_x, input_y = self._consolidate_inputs(x_inputs, y_input, X, y)
+            input_x, input_y = self._consolidate_inputs(x_inputs, y_input, X, most_recent_y)
             self.input_feature_names.update({component_name: list(input_x.columns)})
-
             if isinstance(component_instance, Transformer):
                 if fit:
                     output = component_instance.fit_transform(input_x, input_y)
                 else:
                     output = component_instance.transform(input_x, input_y)
                 if isinstance(output, tuple):
                     output_x, output_y = output[0], output[1]
+                    most_recent_y = output_y
```

Comment:
This keeps track of all the changes done to `y`, since not all transformers pass `y`. For example, when we pass a sampling output to the `StandardScaler`, we want the `y` output from the sampler to be passed to the estimator.

Reply:
yep, that makes sense. I'll keep that in mind for the future!